ecopod/sendRawTransactionConditional.md

```diff
@@ -0,0 +1,53 @@
+# Purpose
+
+To fully unlock ERC-4337 on the OP-Stack.
+
+# Summary
+
+By providing an auxilliary transaction submission mechanism, [eth_sendRawTransactionConditional](https://notes.ethereum.org/@yoav/SkaX2lS9j), we can enable Bundlers to submit Entrypoint transactions with stronger guarantees, avoiding costly inadvertent reverts.
+
+We can implement this endpoint in op-geth with an additional layer of external validation to ensure this endpoint is safe from DoS attacks, does not supersede `eth_sendRawTransaction`, and can be deprecated if native account abstraction efforts in the future absolves the need for this endpoint.
+
+**Note:** As a result of [EIP-3074](https://eips.ethereum.org/EIPS/eip-3074), the interop project is also exploring external validation as a mechanism to preventing DoS attack. A low-stakes scoped version of this for `eth_sendRawTransactionConditional` is a good opportunity for learning.
+
+
+# Problem Statement + Context
+
+Bundlers aggregate UserOps from an external mempool into a single transaction sent to the enshrined 4337 `Entrypoint` contract. It is the Bundler's responsibility to ensure every UserOp in the transaction will not revert when its validation step is executed. In between tx submission and inclusion, there is a chance network state changes causes the transaction to revert, even after local bundler validation.
+
+With private Bundler mempools, the likelihood of these reverts are small. However shared 4337 mempools are coming into [production](https://medium.com/etherspot/decentralized-future-erc-4337-shared-mempool-launches-on-ethereum-b6c860072f41), launched on Ethereum and some L2 testnets, decentralizing 4337 infrastructure. This is possible with no change on Ethereum through special block builders like Flashbots which support atomic transaction inclusion. An endpoint like `eth_sendRawTransactionConditional` is required to launch a shared mempools on L2 as there's only a single block builder, the sequencer, and the likelihood of reverts with a shared mempool would be too high of a cost for bundlers to operate.
+
+
+# Alternatives Considered
+
+## Continue with private 4337 mempools
+
```

Comment:
I don't follow how continuing to use private 4337 mempools is an alternative to implementing the RPC. Its not up to us to determine where the orderflow ends up. Some apps may not want their orderflow to go to a public mempool. I think the alternative is "do nothing"

Reply:
"Do nothing" is likely a better paraphrasing. It's not up to use to determine orderflow but we could still rely on & even push for usage of the existing Bundler stack when when public mempools are live. Relying on a partner like Alchemy to provide vendor-specific AA entrypoints.